Entity/moving_entity.cpp

```diff
@@ -0,0 +1,24 @@
+#include "moving_entity.h"
+
+glm::vec3 MovingEntity::GetVelocity() const {
+    return m_Velocity;
+}
+
+glm::vec3 MovingEntity::GetForward() const {
+    return m_Forward;
+}
+
+glm::vec3 MovingEntity::GetSide() const {
+    return m_Side;
+}
+
+glm::vec3 MovingEntity::GetUp() const {
+    return m_Up;
+}
+float MovingEntity::GetMaxForce() const {
+    return m_MaxForce;
+}
+
+float MovingEntity::GetMaxSpeed() const {
+    return m_MaxSpeed;
```

Comment:
should we make them public then as well? not a fan of writing a lot of getters anyway... @michaeleggers

Reply:
I would say so. Data hiding only really makes sense imo, if:
- The data can be read but cannot be written.
- Setting a variable causes some other sideffect within the class instance or requires some logic to get right.
I have never experienced an advantage of setters / getters if data is literally just read / written anyways other than just a lot of lines of extra code that actually doesn't do anything useful.